conformance/lang/expressions/object-constructor-expr/object_remote_methods.balt

```diff
@@ -0,0 +1,348 @@
+Description: Test method-defn-body satisfying the requirements for an isolated function if isolated-qual is present.
+Labels: additive-expr, array-type, field-access-expr, int, intersection-type, list-constructor-expr, member-access-expr,
+        method-call-expr, remote-method-defn, module-init-var-decl, object-constructor-expr, object-field, readonly-type, var
+
+int[] & readonly arr = [3, 4];
+
+function errorFunction() {
+    var _ = client object {
+        remote isolated function func1() returns int {
+            return arr[1]; // @error invalid access of mutable storage in an isolated function
+        }
+    };
+}
+
+Test-Case: output
+Description: Test method-defn-body satisfying the requirements for an isolated function if isolated-qual is present.
+Labels: additive-expr, array-type, client-remote-method-call-action, int, isolated-function, list-constructor-expr,
+        lock-stmt, member-access-expr, method-call-expr, remote-method-defn, module-init-var-decl, object-constructor-expr,
+        object-field, var
+
+isolated int[] arr = [3, 4];
+
+function init() {
+    var obj = client object {
+        remote isolated function func1() returns int {
+            lock {
+                arr[2] = 5;
+                return arr[0] + arr[2];
+            }
+        }
+    };
+
+    int res = obj->func1();
+    io:println(res); // @output 8
+}
+
+Test-Case: error
+Description: Test method-defn-body satisfying the requirements for an isolated function if isolated-qual is present.
+Labels: int, isolated-function, remote-method-defn, module-init-var-decl, object-constructor-expr, object-field, var
+
+int intVal = 10;
+
+function errorFunction() {
+    var _ = client object {
+        remote isolated function func1() {
+            lock {
+                intVal = 5; // @error invalid access of mutable storage in an isolated function
+            }
+        }
+    };
+}
+
+Test-Case: output
+Description: Test method-defn-body satisfying the requirements for an isolated function if isolated-qual is present.
+Labels: client-remote-method-call-action, explicit-anonymous-function-expr, int, method-call-expr, remote-method-defn,
+        object-constructor-expr, object-field, var
+
+function init() {
+    final int val = 10;
+    var obj = client object {
+        remote isolated function func1() returns int => val;
+    };
+
+    int res = obj->func1();
+    io:println(res); // @output 10
+}
+
+Test-Case: error
+Description: Test method-defn-body satisfying the requirements for an isolated function if isolated-qual is present.
+Labels: explicit-anonymous-function-expr, int, intersection-type, isolated-function, method-call-expr, remote-method-defn,
+        object-constructor-expr, object-field, readonly-type, var
+
+function errorFunction() {
+    int & readonly val = 10;
+    var _ = client object {
+        remote isolated function func1() returns int => val; // @error invalid access of mutable storage in an isolated function
+    };
+}
```

Comment:
Tests to cover 
> The remote-method-name of a remote-method-decl must be distinct from the remote-method-name of every other remote-method-decl in the object-type-descriptor. However, a method-decl and a remote-method-decl can have the same name.

Reply:
Seems latter parts of the spec under `remote-method-decl`  not covered. Can you please check

Reply:
Any reason these should be covered here? This will be covered when adding tests for `object-type-descriptor`. 
These tests only cover `6.8.1.1 Fields` and `6.8.1.2 Methods` sections under `object-constructor-expr` (https://ballerina.io/spec/lang/master/#object-constructor-expr)

Reply:
Any reason, why these should be covered here? This will be covered when adding tests for `object-type-descriptor`, right? :thinking: 
These tests only cover `6.8.1.1 Fields` and `6.8.1.2 Methods` sections under `object-constructor-expr`(https://ballerina.io/spec/lang/master/#object-constructor-expr)

Reply:
My bad. Didn't notice that I was checking a separate section